miio/gateway.py

```diff
@@ -452,8 +471,20 @@ def set_default_music(self):
 class GatewayLight(Device):
     """Light controls for the gateway."""
 
-    def __init__(self, parent) -> None:
-        self._device = parent
+    def __init__(
+        self,
+        parent: Gateway = None,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        if parent is not None:
+            self._gateway = parent
+        else:
+            self._gateway = Device(ip, token, start_id, debug, lazy_discover)
+            _LOGGER.debug("Creating new device instance, only use this for cli interface")
```

Comment:
Considering this gets repeated for every "control" interface, maybe add a new `GatewayDevice` class which has the modified `__init__`. Avoids copy&paste.

Reply:
thanks, have implemented it now